Extensions/ShareTo/InitialViewController.swift

```diff
@@ -48,21 +52,22 @@ class EmbeddedNavController {
             make.center.equalToSuperview()
             make.width.equalTo(width)
             heightConstraint = make.height.equalTo(UX.topViewHeight).constraint
-            if isLandscapeSmallScreen(navigationController.traitCollection) {
-                layout(forTraitCollection: navigationController.traitCollection)
-            }
+            layout(forTraitCollection: navigationController.traitCollection)
         }
 
         navigationController.view.layer.cornerRadius = UX.dialogCornerRadius
         navigationController.view.layer.masksToBounds = true
     }
 
     func layout(forTraitCollection: UITraitCollection) {
-        if isLandscapeSmallScreen(forTraitCollection) {
-            heightConstraint.update(offset: UX.topViewHeight - (UX.numberOfActionRows + 2) * UX.perRowShrinkageForLandscape)
+        let updatedHeight: Int
+        if UX.enableResizeRowsForSmallScreens {
+            let shrinkage = UX.navBarLandscapeShrinkage + (UX.numberOfActionRows + 1 /*one info row*/) * UX.perRowShrinkageForLandscape
+            updatedHeight = isLandscapeSmallScreen(forTraitCollection) ? UX.topViewHeight - shrinkage : UX.topViewHeight
         } else {
-            heightConstraint.update(offset: UX.topViewHeight)
+            updatedHeight = forTraitCollection.verticalSizeClass == .compact ? UX.topViewHeight - UX.navBarLandscapeShrinkage :  UX.topViewHeight
```

Comment:
I snuck in a fix for landscape mode. iOS automatically shrinks navigation bars by 10px in compact, I wasn't handling this properly before.